ovn-org/ovn-kubernetes
# ovn-node: Use workqueues

**- What this PR does and why is it needed**


Changes the ovn node controller from using `WatchFactory` to using workqueues from the client-go package. This builds on the foundation laid in #1314 

**- Special notes for reviewers**

I had to refactor a lot of how the management ports, shared/local gateways etc.. were spawning go routines so everything launches from the central `Run` method. While `Run` is quite nice, there appears to be a lot of overlap in the healtcheck/mgmt port check/openflow check go routines and it probably should be consolidated. I viewed that as out-of-scope for getting this change in

**- How to verify it**

All unit tests passing locally, behaviour should be as before.
But with the following caveats as always.

- There is no Update, Only Add - ergo Add must be idempotent.
- There is no Sync function as the Informer sends events initially and cad do again by adjusting the re-sync interval if required.

**- Description for the changelog**
